### ISSUE TITLE
spec/services/update_status_service_spec.rb:144 is flakey

### ISSUE THREAD

**Issue by shageman** [mastodon/mastodon#24600]
### Steps to reproduce the problem

Run this many times and you will see it fail occasionally
```
bundle exec rspec spec/services/update_status_service_spec.rb:144
```

### Expected behaviour

The test should always pass

### Actual behaviour

The test sometimes fails

### Detailed description

Given that this reloads a poll and immediately expects a message to have been reenqueued, I suspected a timing issue.

When I change the test to the following naive solution, it seems to always pass (or better: pass much more often... to where I didn't see it fail anymore).

```
    it 'requeues expiration notification' do
      poll = status.poll.reload
      sleep 1
      expect(PollExpirationNotifyWorker).to have_enqueued_sidekiq_job(poll.id).at(poll.expires_at + 5.minutes)
    end
```

Making this an issue, because I don't believe this is a good fix to the problem.

### Specifications

mastodon e798b86

**Comment by ClearlyClaire** [comment]
I've ran into this, this is because of sidekiq-unique-jobs but I haven't managed to find a fix.

**Comment by c960657** [comment]
I cannot reproduce this.

@shageman How often does the error occur? How exactly are do you the test?

**Comment by shageman** [comment]
The I filed this issue it was happening about 25% of the time. Today, I can't seem to reproduce it (neither with the sha I used that day or HEAD). 

I am running the entire test suite directly on my machine. All tests pass, I have seen this one spec fail many times, but not always. To confirm the problem for this bug report, I queued up 100s of `bundle exec rspec spec/services/update_status_service_spec.rb:144` and ran them consecutively.
